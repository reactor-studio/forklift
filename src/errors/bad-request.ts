import ForkliftError from './forklift-error';

export default class BadRequestError extends ForkliftError {
  constructor(message: string) {
    super(message, 400, 'Bad request');
  }
}
