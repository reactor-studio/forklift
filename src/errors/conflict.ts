import ForkliftError from './forklift-error';

export default class ConflictError extends ForkliftError {
  constructor(message: string) {
    super(message, 409, 'Conflict');
  }
}
