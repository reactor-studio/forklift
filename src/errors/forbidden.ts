import ForkliftError from './forklift-error';

export default class ForbiddenError extends ForkliftError {
  constructor(message: string) {
    super(message, 403, 'Forbidden');
  }
}
