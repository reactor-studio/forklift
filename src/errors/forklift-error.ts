export default class ForkliftError extends Error {
  readonly status: number;

  readonly name: string;

  readonly stack: string;

  /**
   * Custom error type which one should extend in order to support custom error types,
   * which include custom names and response statuses. The extended error is expected
   * to provide the params in the constructor.
   * @param message
   * @param status
   * @param name
   */
  constructor(message: string, status: number, name: string) {
    super(message);
    this.message = message;
    this.status = status;
    this.name = name;
    Object.setPrototypeOf(this, new.target.prototype);
    Error.captureStackTrace(this, this.constructor);
  }

  toJson = (showTrace = false): object => {
    return {
      status: this.status,
      title: this.name,
      message: this.message,
      meta: showTrace
        ? {
            trace: this.stack,
          }
        : null,
    };
  };
}
