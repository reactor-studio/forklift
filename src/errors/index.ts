export { default as BadRequestError } from './bad-request';
export { default as ForbiddenError } from './forbidden';
export { default as NotFoundError } from './not-found';
export { default as ConflictError } from './conflict';
export { default as ForkliftError } from './forklift-error';
export { default as InputError } from './input-error';
export { default as OutputError } from './output-error';
