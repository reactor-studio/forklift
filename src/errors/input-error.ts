import IoError, { ErrorDetails } from './io-error';

export default class InputError extends IoError {
  constructor(message: string, details?: ErrorDetails) {
    super(message, 'InputError', 400, 'Bad request', details);
  }
}
