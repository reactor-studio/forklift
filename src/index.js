import IO from './io';

export { IoError, ioErrorAdapter } from './io-error';
export { Status } from './status';
export { setLocals, getLocals } from './locals';
export { IO };
