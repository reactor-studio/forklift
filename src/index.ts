export { IoReqError, IoResError, ioErrorAdapter } from './io-error';
export { Status, statusOptions } from './status';
export { setLocals, getLocals } from './locals';
import { IO } from './io';

export default IO;
