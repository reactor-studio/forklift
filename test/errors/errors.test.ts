import { ForkliftError } from '../../src/errors';

describe('Forklift error module', () => {
  it('should provide easy extension of ForkliftError', () => {
    class CustomError extends ForkliftError {
      constructor(message: string) {
        super(message, 999, 'Custom error');
      }
    }

    const test = (): void => {
      throw new CustomError('Some test message');
    };

    try {
      test();
    } catch (e) {
      expect(e.toJson(false)).toMatchSnapshot();
    }
  });
});
