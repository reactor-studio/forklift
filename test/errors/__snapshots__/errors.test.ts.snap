// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forklift error module should provide easy extension of ForkliftError 1`] = `
Object {
  "message": "Some test message",
  "meta": null,
  "status": 999,
  "title": "Custom error",
}
`;
